proxy/api/src/notification.rs

```diff
@@ -43,6 +46,9 @@ impl Subscriptions {
         let id = self.next_id.fetch_add(1, Ordering::Relaxed);
         let (sender, receiver) = mpsc::unbounded_channel();
 
+        if let Some(last_state) = &*self.last_state.read().await {
+            sender.send(last_state.clone()).ok();
```

Comment:
Here's a thought: the notifications shouldn't cache any state. Instead, it's the responsibility of the downstream component that actually cares about what the last status was. In this case, I feel like storing it in the session might make sense?

Reply:
> Here's a thought: the notifications shouldn't cache any state.

:100: 

> I feel like storing it in the session might make sense?

It would conflate operational concerns with state information revolving around the currently active identity.

Reply:
Ya, fair enough. Then we have some other component that thinks about "liveness". It could even be some sort of a feed :thinking: Akin to a window of events, examples being recent announcements and request transitions.

Reply:
I don't think it's the same kind of component. Statuses don't carry much value once changed. The current status and in some cases the last one are really of interest, it's quite ephemeral in nature.

Reply:
So do you think we should just update the `Context` with the `Status`?

Reply:
If we have a handle on the peer to get its status like introduced in #992 

https://github.com/radicle-dev/radicle-upstream/blob/ea733d5ed0c3ba14e428657f4c54dd6b1dfde03f/proxy/api/src/context.rs#L10-L12

The first message can be constructed from synchronously getting the value a la `ctx.peer_control.current_status()` and all follow-up updates can be pushed through the subscription.

Reply:
This would avoid any extra work to keep the status in context up-to-date.

Reply:
I'm not sure I fully understand. Can you elaborate a bit more for me, please?

Reply:
What we like to get is the following semantic, an SSE subscriber from the UI can set up a new subscription -- come to think of it ideally scoped by status -- where it's guaranteed to see the current status as first message. Now the issue is that we only see subsequent status changes once the subscription is set up. To remedy that the current approach is to cache the latest observed notification, which has its own problems as pointed out up top. Therefore we need a way to get the current status send that as the first message and from there on just map the incoming notification from the `subscribe` stream to sse events. One way is to have a `status` field in `Context` as you describe it, the `Context` is ubiquitous in the filter chains of our glorious API. Yet this begs the question where and how do we keep that status field up-to-date. And we learned with blood, sweat and tears that keeping anything in sync and mutable in the `Context` is big pain in the buttocks.

If we use the `PeerControl` aka the handle onto the peer state, we extend that to offer `current_status()`, which in the same fashion as `request_urn()` uses CSP machinery to get something out or into the consumed `Peer`. Hence my suggestion in https://github.com/radicle-dev/radicle-upstream/pull/1011#pullrequestreview-506363462 to stack this on top of #992 and end up with something like:

``` rust
pub async fn current_peer_status_stream(
    ctx: Context,
    subscriptions: Subscriptions,
) -> Result<impl Reply, Rejection> {
    let current_status = ctx.peer_control.current_status();
    let subscriber = subscriptions.subscribe().await;

    let filter = |notification| match notification {
        Notification::LocalPeerStatus(status) => {
            Some(Ok((sse::event("LOCAL_PEER_STATUS"), sse::json(status))))
        }
        _ => None,
    };

    Ok(sse::reply(
        sse::keep_alive().stream(subscriber.filter_map(filter)),
    ))
}
```